Adds unique tests names validator

lowitea opened the pull request:
closes #3

mcproger commented on flake8_fine_pytest/watchers/unique_test_names.py:
```diff
@@ -0,0 +1,35 @@
+
+            node_name = node.name  # type: ignore
+
+            if node_name not in testcases_names:
```
Why is this construction needed? I think the algorithm can be implemented without `testcases_names` set.

lowitea replied:
I couldn't without testcases_names, but I was able to simplify the algorithm  
https://github.com/best-doctor/flake8-fine-pytest/pull/7/commits/1ac73269540ec98da1dd65bd55b9513b40915a2d

mcproger commented on flake8_fine_pytest/watchers/unique_test_names.py:
```diff
@@ -0,0 +1,35 @@
+            error_msg = self.error_template.format(node_name)
+            self.add_error((node.lineno, 0, error_msg))
+
+    def _is_properly_node(self, node: ast.AST) -> bool:
```
This is copy paste. We use this method already in 3 places and 2 of them have similar code. I think we should move this method to `BaseWatcher`, make it public, and override where we need other logic (like [here](https://github.com/best-doctor/flake8-fine-pytest/blob/9eb2b9f4e7029deac9d33027cbb2bba1d992b938/flake8_fine_pytest/watchers/xfail_until_argument_watcher.py#L72))

mcproger replied:
By the way, we have similar case in [another PR](https://github.com/best-doctor/flake8-fine-pytest/pull/8/files#r529247014). Please, take a look, maybe we will do it earlier.

lowitea replied:
Fixed https://github.com/best-doctor/flake8-fine-pytest/pull/7/commits/35f68b6dca164bf37eacfcd1be458d86c6b142cf

mcproger replied:
Have a [comment](https://github.com/best-doctor/flake8-fine-pytest/pull/7/files#r532197073)

mcproger commented on tests/test_files/test_not_unique.py:
```diff
@@ -0,0 +1,19 @@
+import pytest
+
+
+def test_not_uniq() -> None:
```
No need for type annotation in test files.

lowitea replied:
Fixed

mcproger commented:
@lowitea also please, update README with this feature description.

mcproger commented on flake8_fine_pytest/watchers/unique_test_names.py:
```diff
@@ -0,0 +1,27 @@
+            return
+
+        for node in ast.walk(self.tree):
+            if not self._is_test_function(node):
```
Almost done :) 

Better call `_should_check_node` method in this place. This is because the should_check method tells us whether handle this node or just skip it. By default, this method checks that node is test function.
But it can be something else, for example [here](https://github.com/best-doctor/flake8-fine-pytest/blob/master/flake8_fine_pytest/watchers/xfail_until_argument_watcher.py#L69) we need more complex condition to check. So, we are overriding the default behavior and keeping the methods naming consistency.

lowitea replied:
Done :)

mcproger commented:
@lowitea take a look at [this comment](https://github.com/best-doctor/flake8-fine-pytest/pull/7#issuecomment-732700883), please.

Also, you have a [broken build](https://travis-ci.org/github/best-doctor/flake8-fine-pytest/builds/746497973). Please fix it, before merge.

lowitea commented:
I updated readme and fixed ci. But maybe there is a regression after #8. I am failing tests with examples like this:  
```python3
@ pytest.mark.parametrize ('test', ('test',))
def test_not_uniq_with_decorator (test):
     pass


@ pytest.mark.parametrize ('test', ('test2',))
def test_not_uniq_with_decorator (test):
     pass
```
Fell with error "FP009 test_not_uniq_with_decorator should use fixtures as follows"  
@mcproger look at this please

lowitea commented:
Nevertheless, I fixed the tests. A bug or a feature, but definitely does not apply to the current issue

mcproger commented:
Nice job :)
Thanks for your help

lowitea commented:
Hi @mcproger 
I apologize for writing to you here, but I didn't come up with a better way(
Can I ask you (or someone else from your team) to look at my other pull requests

https://github.com/best-doctor/flake8-annotations-coverage/pull/11
https://github.com/best-doctor/flake8-annotations-coverage/pull/9
https://github.com/best-doctor/flake8-variables-names/pull/8
https://github.com/best-doctor/flake8-annotations-complexity/pull/15

mcproger commented:
Hi @lowitea 

>  I didn't come up with a better way(

You can send me email :)

> Can I ask you (or someone else from your team) to look at my other pull requests

Unfortunately, I am not a collaborator of these projects. I'll ask my team, but I'm afraid it may take a while.

lowitea commented:
@mcproger unfortunately I could not find your email

ok, thanks a lot anyway

mcproger commented:
@lowitea you can see it in my profile – mcproger7@gmail.com

lowitea commented:
> @lowitea you can see it in my profile – mcproger7@gmail.com

Thanks!)
